lib/arraySum.js

```diff
@@ -1,7 +1,12 @@
 // Write a function that sums all the numerical elements of an array
 
 function arraySum(array) {
-  // Write code here
+  var sum = 0;
+  if (array.length == 0) { return -1; }
+  for (var i = 0; i < array.length; i++) {
+    sum = (typeof array[i] == "number") ? sum + array[i] : sum;
+  }
+  return sum;
```

Comment:
You don't have to make this change but you may want to check out `array.reduce`. It's really useful for these types of problems.